### ISSUE TITLE
Make a few exceptions to the standard rules

### ISSUE THREAD

**Issue by zeke** [zeke/standard-markdown#2]
The `standard` linter has a few rules that feel restrictive when applied to javascript blocks in markdown files. While [standardizing electron's docs], these are the rules I came across that I think we could do without:

### Disallow Undeclared Variables ([no-undef])

```js
// `win` is not declared in this block
win.close()
```

### Disallow Unused Variables ([no-unused-vars])

```js
// `BrowserWindow` is declared but not used
const {BrowserWindow} = require('electron')
```

### Disallow Unnecessary Nested Blocks ([no-lone-blocks], [no-labels])

Example https://github.com/electron/electron/pull/6748

```js
// this object is just dangling in space
{
  marginsType: 0,
  printBackground: false
}
```

## How?

I think the way to accomplish this is by forking [eslint-config-standard], then using [standard-engine] to load it. We can use the numerous [standard forks] for reference.

@ferros, @flet: If we only wish to deviate very slightly from standard's defaults, is there a way to codify the differences in a `package.json` stanza? Or is [standard-engine] with a custom eslint package the way to go?

[standardizing electron's docs]: https://github.com/electron/electron/pull/6603
[standard-engine]: https://github.com/flet/standard-engine
[eslint-config-standard]: https://github.com/feross/eslint-config-standard
[standard forks]: https://github.com/feross/awesome-standard#forks
[no-undef]: http://eslint.org/docs/rules/no-undef
[no-unused-vars]: http://eslint.org/docs/rules/no-unused-vars
[no-lone-blocks]: http://eslint.org/docs/rules/no-lone-blocks
[no-labels]: http://eslint.org/docs/2.0.0/rules/no-labels

cc @kevinsawicki @jlord @zcbenz

**Comment by MarshallOfSound** [comment]
@zeke So we don't have to fork things crazily.  Couldn't we "hack" in eslint disable comments to the JS blocks we are parsing.

I.e. We have a code block `block`

```js
const disabledRules = ['no-undef', 'no-unused-vars', 'no-lone-blocks', 'no-labels'];
let block = getCodeBlockThing();
block = `
/* eslint-disable ${disabledRules.join(', ')} */
` + block;
standard.lintText(block)
```

This is untested, just an idea 😄

**Comment by feross** [comment]
I didn't see this until now because you misspelled my name, @zeke 😜 

I think disabling these rules would make `standard-readme` way more useful. Good call! They're the reason I haven't adopted it on my projects yet. 

@MarshallOfSound's suggesting is the best solution IMO. It's easiest and easy to maintain -- will have the exact same rules, dependency versions, etc. as the latest stable `standard`.

---

Another option, but less ideal, is to extend `standard` to layer your rule changes on top. No need to fork `eslint-config-standard` to do this.

You can make a `eslintrc.json` file like this one:

```json
{
  "extends": ["standard", "standard-jsx"],
  "rules": {
    "no-undef": 0,
    "no-unused-vars": 0
  }
}
```

Install all the `standard` [dependencies](https://github.com/feross/standard/blob/master/package.json#L14-L22). Then use the `eslintrc.json` with `standard-engine`.

One big downside is now it's your responsibility to ensure that the package versions are kept up-to-date and match what's in `standard`. You can try just always using the latest versions with `*` and it will probably work fine. But there are some situations when ESLint has a bug where we hold back the semver range to an older version until it's fixed. So you lose that with this approach.

**Comment by zeke** [comment]
> I didn't see this until now because you misspelled my name

Doh! I do know how to spell it. I blame my fingers. 🖖

**Comment by zeke** [comment]
#3 and #6 landed in 1.2.0 🚀 

Thanks for the help, @MarshallOfSound and @feross!
